scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -1909,6 +1909,39 @@ class FormatOps(
     }
   }
 
+  object OptionalBraces {
+    def unapply(ft: FormatToken): Option[(FormatToken, Tree)] = ft.left match {
+      case _ if !dialect.allowSignificantIndentation => None
+      case _: T.Comment | _: T.LeftBrace => None
+      case _: T.Colon => // Optional braces in templates after `:`
+        ft.meta.leftOwner match {
+          case t: Template =>
+            val ok = t.tokens.headOption.contains(ft.left)
+            if (ok) Some((ft, t)) else None
+          case _ => None
+        }
+      case _: T.KwWith => // Optional braces in templates after `with`
+        ft.meta.leftOwner match {
+          case t: Template =>
+            val nextFt = nextNonComment(ft)
+            val ok = nextFt.meta.rightOwner.parent.contains(t.self)
```

Comment:
this is slightly hacky, mostly because it's unintuitive to me why the first `def` in this template is owned by the `self` field of the template, and not the template itself. i wonder if there's some other token possible which would start the template but wouldn't be owned by `self`...

Reply:
i got rid of it, using `templateCurly` instead. `self` is inconsistent; sometimes its tokens include the first body token, sometimes they don't (braces and no self annotation).